src/solver/variable/economy/profitByPlant.h

```diff
@@ -0,0 +1,382 @@
+        {
+            // Reset the values for the current year
+            for (unsigned int i = 0; i != pNbClustersOfArea; ++i)
+                pValuesForTheCurrentYear[numSpace][i].reset();
+
+            // Next variable
+            NextType::yearBegin(year, numSpace);
+        }
+
+        void yearEndBuild(State& state, unsigned int year)
+        {
+            // Next variable
+            NextType::yearEndBuild(state, year);
+        }
+
+        void yearEnd(unsigned int year, unsigned int numSpace)
+        {
+            // Merge all results for all thermal clusters
+            {
+                // to automatically flush the memory from times to times
+                unsigned int autoflush = 5;
+
+                for (unsigned int i = 0; i < pNbClustersOfArea; ++i)
+                {
+                    // Compute all statistics for the current year (daily,weekly,monthly)
+                    pValuesForTheCurrentYear[numSpace][i].computeStatisticsForTheCurrentYear();
+
+                    if (!--autoflush)
+                    {
+                        autoflush = 5;
+                        if (Antares::Memory::swapSupport)
+                            Antares::memory.flushAll();
+                    }
+                }
+            }
+            // Next variable
+            NextType::yearEnd(year, numSpace);
+        }
+
+        void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+            unsigned int nbYearsForCurrentSummary)
+        {
+            for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+            {
+                for (unsigned int i = 0; i < pNbClustersOfArea; ++i)
+                {
+                    // Merge all those values with the global results
+                    AncestorType::pResults[i].merge(numSpaceToYear[numSpace],
+                        pValuesForTheCurrentYear[numSpace][i]);
+                }
+            }
+
+            // Next variable
+            NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+        }
+
+        void hourBegin(unsigned int hourInTheYear)
+        {
+            // Next variable
+            NextType::hourBegin(hourInTheYear);
+        }
+
+        void hourForEachArea(State& state, unsigned int numSpace)
+        {
+            // Next variable
+            NextType::hourForEachArea(state, numSpace);
+        }
+
+        void hourForEachThermalCluster(State& state, unsigned int numSpace)
+        {
+            // Useful local variables
+            double* MarginalCosts = state.hourlyResults->CoutsMarginauxHoraires;
+            auto* cluster = state.thermalCluster;
+            double hourlyClusterProduction = state.thermalClusterProduction;
+            uint hourInTheWeek = state.hourInTheWeek;
+            uint hourInTheYear = state.hourInTheYear;
+            int modulationCostColumn = Data::thermalModulationCost;
+
```

Comment:
Not really necessary... You could use `Data::thermalModulationCost` at line 328.

Reply:
ok, done.